packages/schemas/src/Components/Utilities/Get.php

```diff
@@ -31,4 +35,57 @@ public function __invoke(string | Component $path = '', bool $isAbsolute = false
 
         return $component->getState();
     }
+
+    public function string(string $key, bool $isAbsolute = false): string
+    {
+        try {
+            return (string) ($this($key, $isAbsolute) ?? '');
+        } catch (Throwable) {
+            return '';
+        }
+    }
+
+    public function integer(string $key, bool $isAbsolute = false): int
+    {
+        return (int) $this($key, $isAbsolute);
+    }
+
+    public function float(string $key, bool $isAbsolute = false): float
+    {
+        return (float) $this($key, $isAbsolute);
+    }
+
+    public function boolean(string $key, bool $isAbsolute = false): bool
+    {
+        return (bool) $this($key, $isAbsolute);
+    }
+
+    /** @return array<mixed, mixed> */
+    public function array(string $key, bool $isAbsolute = false): array
+    {
+        return (array) ($this($key, $isAbsolute) ?? []);
+    }
+
+    public function date(string $key, bool $isAbsolute = false): ?CarbonInterface
+    {
+        $state = $this($key, $isAbsolute);
+
+        if (! is_string($state)) {
+            return null;
+        }
+
+        return Date::parse($state);
+    }
+
+    /** @param class-string<BackedEnum> $enumClass */
+    public function enum(string $key, string $enumClass, bool $isAbsolute = false): ?BackedEnum
+    {
+        $state = $this($key, $isAbsolute);
+
+        if (! enum_exists($enumClass) || ! is_string($state)) {
```

Comment:
I think it should error if the enum doesn't exist, and also enums can have int backing values, not just strings

```suggestion
        if (blank($state)) {
```